lib/ansible/galaxy/api.py

```diff
@@ -362,9 +375,10 @@ def _call_galaxy(self, url, args=None, headers=None, method=None, auth_required=
                         res['results'].append(result)
 
                 else:
-                    res = path_cache['results']
+                    res = path_cache.get('results')
 
-                return res
+                if res is not None:
+                    return res
```

Comment:
I confirmed this change isn't necessary now - `ansible-galaxy-collection` tests are passing for me locally, the unhandled exception was causing issues.

Reply:
reverted